include/adc.h

```diff
@@ -27,119 +27,225 @@ extern "C" {
  */
 
 /**
- * @brief ADC driver Sequence entry
- *
- * This structure defines a sequence entry used
- * to define a sample from a specific channel.
+ * @brief ADC sampling definition structure.
  */
-struct adc_seq_entry {
-	/** Clock ticks delay before sampling the ADC. */
-	s32_t sampling_delay;
+struct adc_sampling {
+	/**
+	 * Pointer to a buffer where the samples are to be written.
+	 * It must be of proper size, taking into account the number of samples
+	 * and used ADC resolution.
+	 */
+	void *buffer;
 
-	/** Buffer pointer where the sample is written.*/
-	u8_t *buffer;
+	/**
+	 * Channels to be used for getting consecutive samples.
+	 */
+	const u8_t *channel_ids;
 
-	/** Length of the sampling buffer.*/
-	u32_t buffer_length;
+	/**
+	 * Number of samples to read in the sampling.
+	 */
+	u8_t sample_count;
```

Comment:
Maybe this should be channel_count? It's a little confusing to have "samplings" and "samples" be different things

Reply:
I simplified the structure and used a bit-mask for specifying the channels to be included in the sampling. The previous proposal with the `channel_ids` pointer and `sample_count` was too confusing and probably troublesome in implementation (regarding duplicates, as you pointed out in the comment above, and order of the channels).

My idea was to define a sampling as reading of samples from one or more channels. Such samplings can be then arranged in sequences with optional triggers. I added more comments in the header file and changed some names to hopefully present the idea clearer. I am open to change the terminology to a better one. English is not my native language so the names I propose may not be the best (or even good enough) ones.